### DISCUSSION TITLE
dotnet build passes /p:DebugType to wix.exe which fails if not set to 'full' or 'none'

### DISCUSSION THREAD

**Discussion by mindonwarp** [wixtoolset/issues#7954] [Questions]
I could swear this worked just fine a month ago but today somehow our CD pipeline fails and even building the project locally with the same parameters fails. I don't know whether this is something brought in by .net 8.0.101.

So the story is we have a WiX installer project that references several other C# projects and harvests their output.
We call "dotnet build" on the wixproj alone because its dependencies get built automatically. Now, those same projects should generate .pdb files during "Release" build and those symbols are put on our symbols server. To achieve that we set these flags:

![grafik](https://github.com/wixtoolset/issues/assets/1856574/9a647f67-0749-409d-b21e-7416edc62174)

The problem is that the "/p:DebugType=pdbonly" gets passed to wix.exe as "-pdbType=pdbonly" and the wix CLI fails the parameter validation which breaks the build process. As I sad this seems to be a new issue because the same pipeline (and local builds) ran wonderfully a month ago or so. I'm confused.

I found an old closed similar issue (problem was 'embedded') where @robmen mentioned overriding DebugType in the .wixproj directly but in this case unfortunately it doesn't solve the problem because it never gets to that. The CLI parameter validation fails.

That old issue also mentioned that this was a warning and not an error, and now while I'm writing this it comes to mind that maybe dotnet build defaults to "treat all warnings as errors". I'll have a look at that right away.

Still, it would be nice if wix.exe defaulted to 'full' when it gets an unknown DebugType on the CLI.

Thanks for your awesome work!

**Comment by mindonwarp** [comment]
Passing `-nowarn:WIX0268` or any other warning related switch has no effect on wix.exe 😞 Do these get ignored, unlike DebugType?

![grafik](https://github.com/wixtoolset/issues/assets/1856574/c39d0aba-07cd-4298-bc94-b0765ffec1c5)

**Comment by mindonwarp** [comment]
Oh well, it seems I lied 🙈 The reason it worked before is that the builds ran on a different pipeline that didn't contain these `p:DebugSymbols` and `p:DebugType` switches 🤦 should've looked better.

Anyway, it would still be nice if these switches were actually passed down to msbuild while building the .wixproj dependencies (referenced projects). Now to work aroun this I have to build the *.csproj in a separate step just to generate symbols for them.

**Reply by robmen** [reply]
If you set properties on the command-line in MSBuild they are set globally for all projects.

>  Now to work aroun this I have to build the *.csproj in a separate step just to generate symbols for them.

MSBuild provides a few options you could use to handle this scenario, without having to build them separately. But building only the projects you want affected by the global properties is probably best done separately. Global properties are a very heavy hammer.
